src/lib/bootloader/grub_install.rb

```diff
@@ -64,30 +71,48 @@ def no_device_install?
 
     def target
       @target ||= case Yast::Arch.architecture
-                  when "i386"
-                    if @efi
-                      "i386-efi"
-                    else
-                      "i386-pc"
-                    end
-                  when "x86_64"
-                    if @efi
-                      "x86_64-efi"
-                    else
-                      "i386-pc"
-                    end
-                  when "ppc", "ppc64"
-                    raise "EFI on ppc not supported" if @efi
-                    "powerpc-ieee1275"
-                  when "s390_32", "s390_64"
-                    raise "EFI on s390 not supported" if @efi
-                    "s390x-emu"
-                  when "aarch64"
-                    raise "Only EFI supported on aarch64" unless @efi
-                    "arm64-efi"
+                  when "i386"               then i386_target
+                  when "x86_64"             then x64_target
+                  when "ppc", "ppc64"       then ppc_target
+                  when "s390_32", "s390_64" then s390_target
+                  when "aarch64"            then aarch64_target
                   else
                     raise "unsupported architecture '#{Yast::Arch.architecture}'"
                   end
     end
+
+    def i386_target
+      if @efi
```

Comment:
another option I have in mind is to use constants for EFI and NON_EFI targets and do hash lookup.